ring-core/src/ring/util/response.clj

```diff
@@ -195,20 +216,36 @@
              dir-entry  (.getEntry jar-file (add-ending-slash entry-name))]
          (and dir-entry (.isDirectory dir-entry)))))
 
+(defmethod resource-info :jar
+    [^java.net.URL url]
+    (let [conn (.openConnection url)]
+      {:content        (delay (.getInputStream conn))
+       :content-length (delay
+                         (let [len (.getContentLength conn)]
+                           (if (<= 0 len) len)))
+       :directory?     (delay (jar-directory? conn))
+       :last-modified  (delay
+                         (let [last-mod (.getLastModified conn)]
+                           (if-not (zero? last-mod)
+                             (Date. last-mod))))}))
+
+(defmethod resource-info :default
```

Comment:
That sounds like a problem with Clojure more than Ring. It should be Clojure's responsibility to provide better error messages for multimethods.